lms/lmsweb/views.py

```diff
@@ -405,6 +475,10 @@ def view(solution_id: int, file_id: Optional[int] = None):
         'role': current_user.role.name.lower(),
         'versions': versions,
         'test_results': test_results,
+        'shared': shared,
+        'shareable_solutions': SHAREABLE_SOLUTIONS,
+        'shareable_by_user': solution.is_shared,
```

Comment:
You have the solution context inside the template, we already providing it on line 470.